### DISCUSSION TITLE
Binance Web3 Wallet support

### DISCUSSION THREAD

**Discussion by wangrunlin** [rainbow-me/rainbowkit#1616] [Ideas]
https://www.binance.com/en/web3wallet

**Comment by 0xmkh** [comment]
@wangrunlin Seems like on iOS the app can only be tested in TestFlight and it's not fully released yet.

**Comment by hungdoansy** [comment]
Hi @magiziz, do you have plan to support this soon? The app was officially released now

**Reply by 0xmkh** [reply]
Oh nice! Yeah i'll take a look at it and report back to you when it's ready to go live in RainbowKit 👍

**Reply by hungdoansy** [reply]
Thank you. I've been trying it myself but no luck so far. Here's the docs that I read. Hope you find it useful. https://developers.binance.com/docs/binance-w3w/quick-start#integration-with-wagmi-v2

They didn't publish the source code of npm packages.

**Reply by hungdoansy** [reply]
Hi @magiziz can you DM me via Telegram @hungdoansy? I have some findings that could save you some time

**Reply by hungdoansy** [reply]
```js
private registerEventListeners() {
    this.signClient.on('accountsChanged', (accounts) => {
      this.events.emit('accountsChanged', accounts)
    })
    this.signClient.on('chainChanged', (chainId) => {
      this.httpClient = this.setHttpProvider(normalizeChainId(chainId))
      this.events.emit('chainChanged', chainId)
    })
    this.signClient.on('disconnect', () => {
      this.events.emit('disconnect')
    })
    this.signClient.on('uri_ready', (uri: string) => {
      this.events.emit('uri_ready', uri)
    })
  }
```

This is from Binance Ethereum provider (`node_modules/@binance/w3w-ethereum-provider/src/index.ts`). I think we may need to modify `getWalletConnectUri` to listen to `uri_ready` instead of `display_uri`. Haven't tried myself yet. Will do it later today

**Reply by 0xmkh** [reply]
@hungdoansy Thanks for providing the code. I'm looking into this right now 👍
